### ISSUE TITLE
request help: install failed via source on Mac(2.12.0)

### ISSUE THREAD

**Issue by kwanhur** [apache/apisix#6225]
### Issue description

after `make deps` successfully, but `make install` or `sudo make install` failed.

it seems no authority to mkdir dir `/usr/share/lua` . 
Installed lua with brew, so this related dir should locate at `/usr/local/Cellar/lua/5.1/share` .

```bash
➜  apisix git:(master) sudo make install
Password:
install -d /usr/local/apisix/
install -d /usr/local/apisix/logs/
install -d /usr/local/apisix/conf/cert
install conf/mime.types /usr/local/apisix/conf/mime.types
install conf/config.yaml /usr/local/apisix/conf/config.yaml
install conf/config-default.yaml /usr/local/apisix/conf/config-default.yaml
install conf/debug.yaml /usr/local/apisix/conf/debug.yaml
install conf/cert/* /usr/local/apisix/conf/cert/
install -d /usr/share/lua/5.1/apisix
install: mkdir /usr/share/lua: Operation not permitted
make: *** [install] Error 71
```

### Environment

- apisix version (cmd: `apisix version`): 2.12.0
- OS (cmd: `uname -a`): Darwin Kwanhur-MacBook-Pro.local 20.6.0 Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64 x86_64
- luarocks version, if the issue is about installation (cmd: `luarocks --version`): /usr/local/bin/luarocks 3.7.0

```bash
➜ brew info lua
lua: stable 5.4.3 (bottled)
Powerful, lightweight programming language
https://www.lua.org/
/usr/local/Cellar/lua/5.4.3 (29 files, 746.3KB) *
  Poured from bottle on 2021-09-19 at 21:07:48
From: git://mirrors.ustc.edu.cn/homebrew-core.git/Formula/lua.rb
License: MIT
```

```bash
➜ brew info luarocks
luarocks: stable 3.7.0 (bottled), HEAD
Package manager for the Lua programming language
https://luarocks.org/
/usr/local/Cellar/luarocks/3.7.0 (105 files, 715KB) *
  Poured from bottle on 2021-09-19 at 21:22:43
From: git://mirrors.ustc.edu.cn/homebrew-core.git/Formula/luarocks.rb
License: MIT
```

**Comment by Chever-John** [comment]
Could it be that you don't have operation permissions for the file?
Try "ls -" :)

**Comment by kwanhur** [comment]
> Could it be that you don't have operation permissions for the file( /usr/share/lua)? Try "ls -" :)

1. this dir `/usr/share/lua` not exist
2. it hasn't enough authority, even command with `sudo`
3. on my opinion, it should install at lua share dir which from `brew info lua`, not the fixed dir

**Comment by Chever-John** [comment]
Maybe I have a little difference of understanding with you on `sudo` and I suggest you try `mkdir /usr/share/lua` first.

In addition, I would like to confirm whether the processor of your MBP is M1 arch? 
If so, you can refer to this issues #6099 and this [blog](https://github.com/apache/apisix-website/pull/837).

**Comment by Chever-John** [comment]
That's so terrible, but we've at least ruled out that it's not Apache APISIX's fault.
I suggest that you could close this issue.
Best wish & happy Chinese New Year!

**Comment by tzssangglass** [comment]
In my experience, if you are trying to do source development, after you successfully execute `make deps`, you can execute `. /bin/apisix start|stop|version` under source dir, don't need to `make install`.

**Comment by kwanhur** [comment]
> In my experience, if you are trying to do source development, after you successfully execute `make deps`, you can execute `. /bin/apisix start|stop|version` under source dir, don't need to `make install`.

this scenario just install via source, then provide services for another tool.

**Comment by Chever-John** [comment]
@kwanhur Hi, I would like to ask, what kind of solution did you finally choose for this problem. In addition, can you confirm the problem is SIP. I look forward to hearing from you soon.

**Comment by kwanhur** [comment]
Finally, I installed on local VM.


Kwanhur Huang
TL;DR

**Comment by Chever-John** [comment]
What a shocking new. I took the solution idea( SIP ) that you submitted earlier and found that it did not solve the problem.
I'll try again, hoping there will be good news.

**Comment by LancCJ** [comment]
> In my experience, if you are trying to do source development, after you successfully execute `make deps`, you can execute `. /bin/apisix start|stop|version` under source dir, don't need to `make install`.

it works,haha,tks

**Comment by Chever-John** [comment]
> > In my experience, if you are trying to do source development, after you successfully execute `make deps`, you can execute `. /bin/apisix start|stop|version` under source dir, don't need to `make install`.
> 
> it works,haha,tks

So happy it can help you:0
